### PR TITLE
chore(api): Add heater/shaker serial driver

### PR DESCRIPTION
**Pull request by sfoster1**
This is the serial driver for the new heater/shaker module. It shares many commands with the temperature module but also allows for shaking by setting RPM.

Has yet to be tested on hardware.

### CODE REVIEW

File: api/src/opentrons/drivers/utils.py [hunk 1]
```diff
@@ -43,22 +49,52 @@ def parse_string_value_from_substring(substring: str) -> str:
...
             error_message="Unexpected argument to parse_temperature_response",
             parse_source=temperature_string,
         )
 
 
+def parse_rpm_response(rpm_string: str) -> RPM:
+    """Example input: T:1233 C:212"""
+    data = parse_key_values(rpm_string)
+    try:
+        target: Optional[int] = int(parse_number(data["T"], 0))
```
Comment (amitlissack) [inline api/src/opentrons/drivers/utils.py:84]: Do you care to catch `TypeError` too?

File: api/src/opentrons/drivers/utils.py [hunk 2]
```diff
@@ -77,6 +113,18 @@ def parse_plate_temperature_response(
         )
 
 
+def parse_hs_device_information(device_info_string: str) -> Dict[str, str]:
```
Comment (amitlissack) [inline api/src/opentrons/drivers/utils.py:128]: Docstring that is about heater shaker please.

**Review by amitlissack — APPROVED** [review]
Looking good.

Minor nitpicks only.

File: api/src/opentrons/drivers/utils.py (hunk 1 above)
Comment (sfoster1) [reply api/src/opentrons/drivers/utils.py:84]: to catch a failure to coerce to int? isn't that a ValueError?

```
>>> int('24.2')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '24.2'
```

**Review by sotaoverride — APPROVED** [review]
lgtm

File: api/src/opentrons/drivers/utils.py (hunk 1 above)
Comment (amitlissack) [reply api/src/opentrons/drivers/utils.py:84]: Yes. My mistake.

File: api/src/opentrons/drivers/utils.py [hunk 3]
```diff
@@ -43,22 +49,52 @@ def parse_string_value_from_substring(substring: str) -> str:
 
 
 def parse_temperature_response(
-    temperature_string: str, rounding_val: int
+    temperature_string: str, rounding_val: int, zero_target_is_unset: bool = False
```
Comment (DerekMaggio) [inline api/src/opentrons/drivers/utils.py:52]: A docstring explaining what `zero_target_is_unset` means would be helpful

**Review by DerekMaggio — APPROVED** [review]

**Pull request merged**